cryptoprotocol/src/main/java/de/upb/crypto/incentive/cryptoprotocol/exceptions/PedersenException.java

```diff
@@ -0,0 +1,10 @@
+package de.upb.crypto.incentive.cryptoprotocol.exceptions;
+
+public class PedersenException extends Exception
```

Comment:
If you make this a `RuntimeException` instead of an `Exception`, you won't be forced to build `try/catch` around everything that can throw this exception. 
Our internal rule for this is roughly: if the error can be avoided by the programmer supplying normal input (e.g., a `divide(5,3)` call shouldn't force me to plan for the case that one of the parameters is zero. I know what I've put there and this case cannot happen), then we generally use `RuntimeException`s (so that our code doesn't become a giant `try/catch` fest). 
If an error is kind of unavoidable from the programmer's side (e.g., reading stuff from an internet socket whose connection can fail at any time even if the programmer's work is perfect), use `Exception`.

Another thing to think about: often, you don't actually need special custom `Exception`s - in my experience, a simple standard `IllegalArgumentException` is just as expressive (given the right message).

Reply:
Pedersen exception class is gone. Exceptions that are almost-surely prevented from the programmer's site are now just thrown and not handled, as suggested